src/typecheck.c

```diff
@@ -779,17 +779,26 @@ static const Type *typecheck_function_or_method_call(FileTypes *ft, const AstCal
     char *sigstr = signature_to_string(sig, false);
 
     int n = call->nargs + !!self_type;
-    if (n < sig->nargs || (n > sig->nargs && !sig->takes_varargs)) {
-        fail_with_error(
-            location,
-            "%s %s takes %d argument%s, but it was called with %d argument%s",
-            self_type ? "method" : "function",
-            sigstr,
-            sig->nargs,
-            sig->nargs==1?"":"s",
-            call->nargs,
-            call->nargs==1?"":"s"
-        );
+    if (n < sig->nargs || (n > sig->nargs && !sig->takes_varargs)) { // Check if it is a function in the class
+        if (self_type->kind != TYPE_CLASS) {
```

Comment:
It is segfaulting because `self_type` can be NULL here. It means that you have a function, not a method.

But as I said, typecheck is not the right place to detect a missing `self` argument. It should be done already when parsing.

Reply:
changed check in parse